lib/identity_cache/cached/attribute.rb

```diff
@@ -90,42 +46,38 @@ def expire(record)
         end
       end
 
-      def input_key_to_cache_key(key_values)
-        values_hash = IdentityCache.memcache_hash(unhashed_values_cache_key_string(key_values))
+      def cache_key(index_key)
+        values_hash = IdentityCache.memcache_hash(unhashed_values_cache_key_string(index_key))
         "#{model.rails_cache_key_namespace}#{cache_key_prefix}#{values_hash}"
       end
 
-      private
-
-      def load_one_from_db(key_values)
-        query = model.reorder(nil).where(Hash[key_fields.zip(key_values)])
+      def load_one_from_db(key)
+        query = model.reorder(nil).where(load_from_db_where_conditions(key))
         query = query.limit(1) if unique
         results = query.pluck(attribute)
         unique ? results.first : results
       end
 
-      def load_multi_from_db(index_values)
-        key_field = key_fields.first
-        rows = model.reorder(nil).where(key_field => index_values).pluck(key_field, attribute)
-        result = {}
-        default = unique ? nil : []
-        index_values.each do |index_value|
-          result[index_value] = default.try!(:dup)
-        end
-        if unique
-          rows.each do |index_value, attribute_value|
-            result[index_value] = attribute_value
-          end
-        else
-          rows.each do |index_value, attribute_value|
-            result[index_value] << attribute_value
-          end
-        end
-        result
+      private
```

Comment:
If we expect these methods to be overridden by the child and they aren't public, should we make them `protected` instead.

Reply:
In ruby, `protected` is to allow them to be accessed from another instance.  If their purpose isn't clear enough, maybe I could make that clearer with documentation.  Looks like YARD has an [`@abstract` tag](https://rubydoc.info/gems/yard/file/docs/Tags.md#abstract) that could be used for that purpose.

Reply:
Yeah, that should work too. I don't typically see abstract methods that are private though I'm sure they exist.